datastore/google/cloud/datastore/transaction.py

```diff
@@ -234,3 +282,36 @@ def commit(self):
         finally:
             # Clear our own ID in case this gets accidentally reused.
             self._id = None
+
+    def put(self, entity):
+        """Remember an entity's state to be saved during :meth:`commit`.
+
+        .. note::
+           Any existing properties for the entity will be replaced by those
+           currently set on this instance.  Already-stored properties which do
+           not correspond to keys set on this instance will be removed from
+           the datastore.
+
+        .. note::
+           Property values which are "text" ('unicode' in Python2, 'str' in
+           Python3) map to 'string_value' in the datastore;  values which are
+           "bytes" ('str' in Python2, 'bytes' in Python3) map to 'blob_value'.
+
+        When an entity has a partial key, calling :meth:`commit` sends it as
+        an ``insert`` mutation and the key is completed. On return,
+        the key for the ``entity`` passed in is updated to match the key ID
+        assigned by the server.
+
+        :type entity: :class:`google.cloud.datastore.entity.Entity`
+        :param entity: the entity to be saved.
+
+        :raises: :class:`RuntimeError` if the transaction
+                 is marked ReadOnly
+                 :class:`ValueError` if the batch is not in
+                 progress, if entity has no key assigned, or if the key's
+                 ``project`` does not match ours.
+        """
+        if isinstance(self.options.mode, self.options.ReadWrite):
+            super(Transaction, self).put(entity)
+        else:
+            raise RuntimeError("Transaction is read only")
```

Comment:
I'd prefer a `ValueError` here, `RuntimeError` is reserved for "we don't really know what to call it".

Reply:
I thought about ValueError, but it's not that I'm passing a bad value to put.  But I shouldn't be able to use this function no matter what value I give it.  I didn't think it fit any other standard errors, and IOError doesn't seem right, thus RuntimeError was the best I thought.  I don't mind changing it though.

Reply:
The "value" in error would be the value of `self.options` (or `self._read_only` if you go that route).